### ISSUE TITLE
Exception when adding multiple PolicyHandlers to IHttpClientBuilder

### ISSUE THREAD

**Issue by angelaki** [dotnet/runtime#61056]
### Description

With the current .net Version 6.0 I get this exception on execution when adding multiple PolicyHandlers:

An item with the same key has already been added. Key: PolicyHttpMessageHandler.PriorResponse.

### Reproduction Steps

```
    .AddPolicyHandler(GetRetryPolicy())
    .AddPolicyHandler(GetTimeoutPolicy())

    public IAsyncPolicy<HttpResponseMessage> GetRetryPolicy(int retryCount = 5)
        => HttpPolicyExtensions
            .HandleTransientHttpError()
            .Or<TimeoutRejectedException>()
            .OrResult(msg => msg.StatusCode == System.Net.HttpStatusCode.NotFound)
            .OrResult(msg => msg.StatusCode == System.Net.HttpStatusCode.Unauthorized)
            .WaitAndRetryAsync(retryCount, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt - 1)))
        ;

    public IAsyncPolicy<HttpResponseMessage> GetTimeoutPolicy(TimeSpan? timeout = null)
        => Policy.TimeoutAsync<HttpResponseMessage>(timeout ?? TimeSpan.FromSeconds(100));
```

### Expected behavior

Was working with 5.0

### Actual behavior

Exception `An item with the same key has already been added. Key: PolicyHttpMessageHandler.PriorResponse.`

### Regression?

Had no issues with 5.0

**Comment by angelaki** [comment]
You can also you double-add e.g. `.AddPolicyHandler(GetTimeoutPolicy())`, so there is no Polly dependency.

**Comment by CarnaViire** [comment]
Thanks @angelaki. This looks like a duplicate of https://github.com/dotnet/aspnetcore/issues/37547 which is already fixed and backported to 6.0.
